closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -361,22 +365,20 @@ int encryptFinal(byte[] in, int inOfs, int len, byte[] out, int outOfs)
 
         // refresh 'in' to all buffered-up bytes
         in = ibuffer_enc.toByteArray();
-        inOfs = 0;
-        len = in.length;
         ibuffer_enc.reset();
 
         byte[] aad = ((aadBuffer == null || aadBuffer.size() == 0) ? emptyAAD : aadBuffer.toByteArray());
 
-        int ret = nativeCrypto.GCMEncrypt(key, key.length,
-                iv, iv.length,
-                in, inOfs, len,
+        int ret = nativeCrypto.GCMEncrypt(keyCopy, keyCopy.length,
+                ivCopy, ivCopy.length,
+                in, 0, in.length,
                 out, outOfs,
                 aad, aad.length, tagLenBytes);
         if (ret == -1) {
             throw new ProviderException("Error in Native GaloisCounterMode");
         }
 
-        return (len + tagLenBytes);
+        return (in.length + tagLenBytes);
```

Comment:
Please no parentheses here (`return` is not a function).

Reply:
Fixed.